10-gis.Rmd

```diff
@@ -895,6 +897,70 @@ But the same is true for the lightweight SQLite/SpatiaLite database engine and G
+
+For larger areas of interest, requested images are still relatively difficult to work with: they may use different map projections, may spatially overlap, and the spatial resolution often depends on the spectral band. The **gdalcubes** package [@appel_gdalcubes_2019] can be used to abstract from individual images and to create and process image collections as four-dimensional data cubes.
+
+The example below shows a minimal example to create a lower resolution (250m) maximum NDVI composite from the Sentinel-2 images returned by the previous STAC-API search.
+
+```{r 09-gdalcubes-example, eval = FALSE}
+library(gdalcubes)
+
+# Create an image collection object from STAC items
+collection = stac_image_collection(items$features, property_filter = function(x) {x[["eo:cloud_cover"]] < 10})
+
+# Define spatiotemporal extent, resolution and CRS of the target data cube
+v = cube_view(srs = "EPSG:3857", extent = collection, dx = 250, dy = 250, dt = "P1D")
+
+# Create and process the data cube
+raster_cube(collection, v) |>
+  select_bands(c("B04", "B08")) |>
+  apply_pixel("(B08-B04)/(B08+B04)", "NDVI") |>
+  reduce_time("max(NDVI)") -> cube
+
+# gdalcubes_options(parallel = 8)
+# plot(cube, zlim = c(0,1))
+```
+
+The combination of STAC, COGs, and data cubes forms a cloud-native workflow to analyze (larger) collections of satellite imagery in the cloud. For more details, please refer to this [tutorial presented at OpenGeoHub summer school 2021](https://appelmar.github.io/ogh2021/tutorial.html).
+
+
+### openEO
+
+Besides hosting large data archives, numerous cloud-based services to process Earth observation data have been launched during the last years. OpenEO [@schramm_openeo_2021] is an initiative to support interoperability among cloud services by defining a common language for processing the data. The initial idea has been described in an [r-spatial.org blog post](https://r-spatial.org/2016/11/29/openeo.html) and aims at making it possible for users to change between cloud services easily with as little code changes as possible. The [standardized processes](https://processes.openeo.org) use a multidimensional data cube model as an interface to the data. Implementations are available for eight different backends (see https://hub.openeo.org) to which users can connect with R, Python, JavaScript, QGIS, or a web editor and define (and chain) processes on collections. Since the functionality and data availability differs among the backends, the **openeo** R package [@lahn_openeo_2021] dynamically loads available processes and collections from the connected backend. Afterwards, users can load image collections, apply and chain processes, submit jobs, and explore and plot results. 
```

Comment:
@appelmar can you think of a small code example for this section? I think it would be great to see openeo in action (if possible).

Reply:
AFAIK, all available services require a user account. Do you think it would be useful to include an example without connecting to a specific backend?

Reply:
Yes -- it could help people to do their first steps (after getting the user account).

Reply:
I've added an example, hope this is useful and easy to understand?